lib/ace/edit_session/bracket_match.js

```diff
@@ -37,17 +37,76 @@ var Range = require("../range").Range;
 
 function BracketMatch() {
 
+
+    this.findSurroundingBrackets = function(position) {
+        if ((position.column == 0) && (position.row)) return null;
+
+        var allBrackets = { 
+            left : [this.$findOpeningBracket("}", position, /(\.?.paren)+/), 
+                    this.$findOpeningBracket(']', position, /(\.?.paren)+/), 
+                    this.$findOpeningBracket(')', position, /(\.?.paren)+/)],
+            right : [this.$findClosingBracket("{", position, /(\.?.paren)+/),
+                     this.$findClosingBracket('[', position, /(\.?.paren)+/),
+                     this.$findClosingBracket('(', position, /(\.?.paren)+/)]
+        };
+
+        var left = {row: null, column: null};
+        var right = {row: null, column: null};
+
+        var leftRowArr = [],
+            leftColArr = [],
+            rightColArr = [],
+            rightRowArr = [];
+
+        var length = Math.max(allBrackets.left.length, allBrackets.right.length)
```

Comment:
really? I think that in this case some element in the array will be null (and you check this below), but arrays will anyway have the same length.Even if you have no brackets at all and they both contain three nulls, the length will be 3. Anything that I am missing?